packages/x-tree-view/src/internals/plugins/useTreeViewNodes/useTreeViewNodes.types.ts

```diff
@@ -66,9 +70,12 @@ export interface TreeViewNodeIdAndChildren {
   children?: TreeViewNodeIdAndChildren[];
 }
 
-export interface UseTreeViewNodesState {
-  nodeTree: TreeViewNodeIdAndChildren[];
-  nodeMap: TreeViewNodeMap;
+export interface UseTreeViewNodesState<R extends {}> {
+  nodes: {
```

Comment:
I scoped the node state because having 3 elements at the root was starting to be messy